src/net/sourceforge/kolmafia/maximizer/Evaluator.java

```diff
@@ -2039,7 +2039,7 @@ else if ( spec.compareTo( secondBest ) > 0 )
 			// If we currently have nothing equipped, also consider leaving nothing equipped
 			if ( EquipmentManager.getEquipment( Evaluator.toUseSlot( slot ) ) == EquipmentRequest.UNEQUIP )
 			{
-				ranked[ slot ].add( new CheckedItem( 0, equipScope, maxPrice, priceLevel ) );
+				ranked[ slot ].add( new CheckedItem( -1, equipScope, maxPrice, priceLevel ) );
```

Comment:
Since I am not viewing this in a rich environment, is this the only place CheckedItem is called?  If it isn't then it might be safer to replace the check in the above with something that captured all invalid itemIds and mapped them to NONE.  I don't know how the Mazimizer might get presented with a pseudo Item or Item 13 but if it can be ruled out....

Reply:
There are a few other constructor invocations in Evaluator.

I missed another instance where itemId was specified as 0. I believe all other instances are receiving itemIds from actual items in inventory.

I wouldn't typically object to making this a catch-all for "hey this isn't a real item", although it may lead to unnecessary checks which may lead to Maximizer slowdowns, depending on how we do this.

The robust approach would be to check ItemDatabase to confirm that the itemId doesn't map to a real item. Each new CheckedItem would spawn a HashMap lookup.

The quick-and-dirty approach would be to check that itemId <= 0.